components/api/src/main/java/com/hotels/styx/api/extension/service/ConnectionPoolSettings.java

```diff
@@ -44,12 +41,12 @@
                            @Deprecated Integer socketTimeoutMillis,
                            Integer pendingConnectionTimeoutMillis,
```

Comment:
I would just add a private method in this file:

```    
    private <T> T firstNonNull(T a, T b) {
        return a != null ? a : b;
    }
```

Just to avoid accidentally changing the initialisation logic.

Reply:
... although having seen @Noddy76 alternative below, I will use that instead.